lib/rack/cache/meta_store.rb

```diff
@@ -165,6 +165,17 @@ def requests_match?(vary, env1, env2)
       end
     end
 
+    # Retrieve the literal value of the Vary header from a cached response hash
+    # without case sensitivity.
+    def vary(hash)
+      hash[hash_key(hash, 'Vary')]
```

Comment:
Probably a more appropriate way to do this is by utilizing `Rack::Utils::HeaderHash`:
```ruby
Rack::Utils::HeaderHash.new(hash)['Vary']
```

but it is a little slow..

```ruby
def vary(hash)
  hash[hash_key(hash, 'Vary')]
end

def hash_key(hash, key)
  hash.each_key { |k| return k if k.casecmp(key).zero? }
  nil
end

def vary_header_hash(hash)
  Rack::Utils::HeaderHash.new(hash)['Vary']
end

headers = {
  "Access-Control-Allow-Origin" => "*",
  "Connection" => "Keep-Alive",
  "Content-Encoding" => "gzip",
  "Content-Type" => "text/html; charset=utf-8",
  "Date" => "Mon, 18 Jul 2016 16:06:00 GMT",
  "Etag" => '"c561c68d0ba92bbeb8b0f612a9199f722e3a621a"',
  "Keep-Alive" => "timeout=5, max=997",
  "Last-Modified" => "Mon, 18 Jul 2016 02:36:04 GMT",
  "Server" => "Apache",
  "Set-Cookie" => "mykey=myvalue; expires=Mon, 17-Jul-2017 16:06:00 GMT; Max-Age=31449600; Path=/; secure",
  "Transfer-Encoding" => "chunked",
  "Vary" => "Cookie, Accept-Encoding",
  "X-Backend-Server" => "developer2.webapp.scl3.mozilla.com",
  "X-Cache-Info" => "not cacheable; meta data too large",
  "X-kuma-revision" => "1085259",
  "x-frame-options" => "DENY"
}

Benchmark.ips do |bm|
  bm.report("vary") { vary(headers) }
  bm.report("HeaderHash") { vary_header_hash(headers) }
  bm.compare!
end
```

```
Warming up --------------------------------------
                vary
   269.993k i/100ms
          HeaderHash    43.139k i/100ms
Calculating -------------------------------------
                vary      2.762M (± 1.0%) i/s -     14.040M in   5.083923s
          HeaderHash    460.732k (± 3.2%) i/s -      2.330M in   5.061061s

Comparison:
                vary:  2761864.9 i/s
          HeaderHash:   460731.6 i/s - 5.99x  (± 0.00) slower
```